### PR TITLE
ts-migration/valid-expect

### PR DESCRIPTION
**Pull request by G-Rath**
chore(valid-expect): convert to typescript

Closes #333

### CODE REVIEW

File: src/rules/__tests__/valid-expect.test.ts [hunk 1]
```diff
@@ -15,23 +15,65 @@ ruleTester.run('valid-expect', rule, {
...
         'test("valid-expect", () => expect(Promise.resolve(2)).resolves.toBeDefined());',
       options: [{ alwaysAwait: true }],
     },
     'test("valid-expect", () => expect(Promise.resolve(2)).resolves.toBeDefined());',
     'test("valid-expect", () => expect(Promise.reject(2)).rejects.toBeDefined());',
-    'test("valid-expect", () => expect(Promise.reject(2)).not.resolves.toBeDefined());',
-    'test("valid-expect", () => expect(Promise.reject(2)).not.rejects.toBeDefined());',
     'test("valid-expect", () => expect(Promise.reject(2)).resolves.not.toBeDefined());',
     'test("valid-expect", () => expect(Promise.reject(2)).rejects.not.toBeDefined());',
+    // 'test("valid-expect", () => expect(Promise.reject(2)).not.resolves.toBeDefined());',
```
Comment (SimenB) [inline src/rules/__tests__/valid-expect.test.ts]: ?

File: src/rules/__tests__/valid-expect.test.ts [hunk 2]
```diff
@@ -15,23 +15,65 @@ ruleTester.run('valid-expect', rule, {
     'expect(undefined).not.toBeDefined();',
     'test("valid-expect", () => { return expect(Promise.resolve(2)).resolves.toBeDefined(); });',
     'test("valid-expect", () => { return expect(Promise.reject(2)).rejects.toBeDefined(); });',
-    'test("valid-expect", () => { return expect(Promise.resolve(2)).not.resolves.toBeDefined(); });',
-    'test("valid-expect", () => { return expect(Promise.resolve(2)).not.rejects.toBeDefined(); });',
-    'test("valid-expect", function () { return expect(Promise.resolve(2)).not.resolves.toBeDefined(); });',
-    'test("valid-expect", function () { return expect(Promise.resolve(2)).not.rejects.toBeDefined(); });',
-    'test("valid-expect", function () { return Promise.resolve(expect(Promise.resolve(2)).not.resolves.toBeDefined()); });',
-    'test("valid-expect", function () { return Promise.resolve(expect(Promise.resolve(2)).not.rejects.toBeDefined()); });',
+    'test("valid-expect", () => { return expect(Promise.resolve(2)).resolves.not.toBeDefined(); });',
```
Comment (SimenB) [inline src/rules/__tests__/valid-expect.test.ts]: would be cool to error on these _not_ being valid. can do that later, though

Comment (G-Rath) [reply src/rules/__tests__/valid-expect.test.ts]: Agreed

File: src/rules/__tests__/valid-expect.test.ts (hunk 1 above)
Comment (G-Rath) [reply src/rules/__tests__/valid-expect.test.ts]: I left there here commented out b/c the whole thing was a mess, and I didn't know what the & wasn't valid :)

Case & point, up until now we were testing the the exact opposite of the correct (`not.resolve`) was correct.

Feel free to remove them if you want, but I'll def be circling back, so they won't be there for ever :)

**Comment by SimenB** [comment]
This turned out pretty clean, I love it! 😀

**Comment by G-Rath** [comment]
Just wait until we've got all the rules converted, and I do my second pass over everything ;)

**Pull request merged**